### PR TITLE
Add logging context information to ledger services

### LINKED ISSUE #3699: Enable usage of structured logging
We already use slf4j as the standard logging interface in the Ledger API Server. We now want to standardize the log messages so they are represented in a structured way. The format we chose is [logfmt](https://www.brandur.org/logfmt). An existing library that seems helpful: [Escalog](https://github.com/BatchLabs/escalog)

The keys used should have a consistent name (i.e. avoid using the various forms of `cmdId`, `commandId`, `commandID`, ...).

* The current log message should be logged with the key `msg`.
* IDs to log include:
  * Party
  * CommandID
  * TransactionID
  * EventID
  * Offset

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Rolls back that AllowedType implicit, which made the logging context
less usable and didn't provide that much of an advantage. Replaced with
helpers where it makes sense that take care of turning the value safely
into a string.

Closes #3699

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala [hunk 1]
```diff
@@ -74,92 +75,86 @@ final class ApiCommandService private (
...
-                      )
-
-                    if (configuration.limitMaxCommandsInFlight)
-                      MaxInFlight(configuration.maxCommandsInFlight).joinMat(tracker)(Keep.right)
-                    else tracker
-                  }
+  private def submitAndWaitInternal(request: SubmitAndWaitRequest): Future[Completion] =
+    withEnrichedLoggingContext(
+      logging.commandId(request.getCommands.commandId),
+      logging.party(request.getCommands.party)) { implicit logCtx =>
```
Comment (SamirTalwar) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: Makes subsequent diffs nicer if we use trailing commas.

```suggestion
      logging.party(request.getCommands.party),
    ) { implicit logCtx =>
```

There's a few other places in the diff too, which you're welcome to address or not as you like.

**Comment by stefanobaghino-da** [comment]
>  because at some point we'll need to log things that aren't best represented as strings
If it can't be "best" represented as a string, I'm not sure it should go in a JSON. :wink:

**Comment by stefanobaghino-da** [comment]
Closed by mistake, I'll put my fingers on a diet. :smile:

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: I get a compilation error if I try to do this, are you sure this is valid syntax for varargs?

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: Addressed by https://github.com/digital-asset/daml/pull/4205/commits/c887bf168c6fd7d8331aeedd61e5c284d1e47ae8

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiCommandService.scala]: > I get a compilation error if I try to do this, are you sure this is valid syntax for varargs?

TIL trailing commas are valid only before a trailing newline. Thanks for clarifying this offline.

**Pull request merged**